src/handlers/backport.rs

```diff
@@ -0,0 +1,241 @@
+            true
+        })
+        .collect();
+    if valid_configs.is_empty() {
+        log::warn!(
+            "Skipping backport nomination: could not find a suitable backport config. Please ensure the triagebot.toml has a `[backport.*.required_pr_labels]` section matching the team label(s) for PR #{}.",
+            pr.number
+        );
+        return Ok(None);
+    }
+
+    // Check marker text in the opening comment of the PR to retrieve the issue(s) being fixed
+    for caps in CLOSES_ISSUE_REGEXP.captures_iter(&event.issue.body) {
+        let id = caps.name("issue_num").unwrap().as_str();
+        let id = match id.parse::<u64>() {
+            Ok(id) => id,
+            Err(err) => {
+                return Err(format!("Failed to parse issue id `{id}`, error: {err}"));
+            }
+        };
+        input.ids.push(id);
+    }
+    log::info!(
+        "Will handle event action {:?} in backport. Regression IDs found {:?}",
+        event.action,
+        input.ids
+    );
+
+    Ok(Some(input))
+}
+
+pub(super) async fn handle_input(
+    ctx: &Context,
+    _config: &BackportTeamConfig,
+    event: &IssuesEvent,
+    input: BackportInput,
+) -> anyhow::Result<()> {
+    let pr = &event.issue;
+
+    // Retrieve the issue(s) this pull request closes
+    let issues = input
+        .ids
+        .iter()
+        .copied()
+        .map(|id| async move { event.repository.get_issue(&ctx.github, id).await });
+
+    // Add backport nomination label to the pull request
+    for issue in issues {
+        let issue = issue.await.unwrap();
+        let mut regression_label = String::new();
+        let issue_labels: Vec<&str> = issue
+            .labels
+            .iter()
+            .map(|l| {
+                // save regression label for later
+                if l.name.starts_with("regression-from-") {
+                    regression_label = l.name.clone();
+                }
+                l.name.as_str()
+            })
+            .collect();
+
+        // Check issue for a prerequisite regression label
+        if regression_label.is_empty() || !contains_any(&REGRESSION_LABELS, &[&regression_label]) {
+            return Ok(());
+        }
+
+        // Check issue for a prerequisite priority label
+        if !contains_any(&issue_labels, &PRIORITY_LABELS) {
+            return Ok(());
+        }
+
+        // figure out the labels to be added according to the regression label
+        let add_labels = input.profile_labels.get(&regression_label);
+        if add_labels.is_none() {
+            log::warn!(
+                "Skipping backport nomination: nothing to do for issue #{}. No config found for regression label ({:?})",
+                issue.number,
+                REGRESSION_LABELS
+            );
+            return Ok(());
+        }
+
+        // Add backport nomination label(s) to PR
+        let mut new_labels = pr.labels().to_owned();
+        new_labels.extend(
+            add_labels
```

Comment:
No need to do that, the `add_labels` call below already appends to the existing list of labels. You can just pass the `add_labels`.

Reply:
The function `add_labels()` wants a Vec<Label>`. The variable `add_labels` is a `Vec<String>` (and I'd prefer leaving it like that for ease of handling)